### PR TITLE
[Flutter-Parent][MBL-13677] Fetch and other tests

### PR DESCRIPTION
**Pull request by CalvinKern**
Also renamed the api directory to network, then making a new directory inside of that called api. This is so the network/utils can be reported via code coverage while keeping network/api excluded.

Also also, a few other tweaks to make logs cleaner and to get coverage higher.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1015664&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/test/network/fetch_test.dart [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      await fetchFirstPage(_requestFail()).catchError((_) {
+        fail = true; // Don't return, just update the flag
+      });
+      expect(fail, isTrue);
+    });
+  });
+
+  // Not able to test getting data for a next page, as we have no way of mocking Dio which is accessed directly in fetch
+  group('fetch next page', () {
+    test('catches errors and returns a Future.error', () async {
```
Comment (joehoag) [inline apps/flutter_parent/test/network/fetch_test.dart]: Maybe this test should be named something else, if we can't really test fetching the data for the next page?

Comment (joehoag) [reply apps/flutter_parent/test/network/fetch_test.dart]: My mistake -- I confused the group name with the test name.

**Pull request merged**